oceannavigator/frontend/src/components/DatasetSelector.jsx

```diff
@@ -244,6 +246,21 @@ class DatasetSelector extends React.Component {
     }
 
     if (this.variableChanged(key)) {
+      if (this.state.datasetDepths.length === 0) {
+        this.setState({ 
+          loading: true, 
+          loadingPercent: 70,  
+          loadingTitle: `${this.state.datasetDepths}` 
+        });
+        GetDepthsPromise(this.state.dataset, value).then(depthResult => {
+          this.setState({
+            datasetDepths: depthResult.data,
+            depth: 0,
+            loading: false,
+            loadingPercent: 0
+          })
+        });
+      }
```

Comment:
I'd suggest moving this into the `changeVariable` function, since this is a side effect of changing a variable.